src/modules/users/users.repository.ts

```diff
@@ -0,0 +1,97 @@
+import { customError } from 'messages/errors';
+import { customErrors } from 'messages/errors';
+import { EntityRepository, Repository } from 'typeorm';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { User } from './entities/user.entity';
+
+@EntityRepository(User)
+export class UsersRepository extends Repository<User> {
+  async findUserByEmail(userEmail: string): Promise<User> {
+    const data = await this.find({ where: { email: userEmail } });
+    const userDB = data[0];
+
+    if (!userDB) {
+      throw new customError(
+        customErrors.EMAIL_IS_INCORRECT.message,
+        customErrors.EMAIL_IS_INCORRECT.code,
+      );
+    }
+
+    return userDB;
+  }
+
+  async readAllUsers(
+    searchParams: any = {},
```

Comment:
function arguments with default value should be marked with optional in TS and should be located after required params

- [ ] add rule to eslint for optional arguments
- [ ] fix style